hack/build-go.sh

```diff
@@ -14,7 +14,8 @@ os::log::install_errexit
 
 # only works on Linux for now, all other platforms must build binaries themselves
 if [[ -z "$@" ]]; then
-  if os::build::detect_local_release_tars $(os::build::host_platform_friendly) >/dev/null; then
+  if [[ "${OS_RELEASE:-}" != "n" ]] && \
```

Comment:
double opt-out feels weird (I call build-go.sh, which detects tar files and opts me out of the build, but I opt out of the opt-out). Also, comparing to "n" is weird... if anything, I'd expect something more like `FORCE_BUILD=any-non-empty-value`

@smarterclayton, what was the use case around build extracting tars, and how would you prefer it be overridden?